content/enterprise_influxdb/v1.9/flux/optimize-queries.md

```diff
@@ -0,0 +1,182 @@
+| **window()** \|> **sum()**     | {{< icon "check" >}} |
+
+{{% caption %}}
+{{< req "\*" >}} **filter()** only pushes down when all parameter values are static.
+See [Avoid processing filters inline](#avoid-processing-filters-inline).
+{{% /caption %}}
+
+Use pushdown functions and function combinations at the beginning of your query.
+Once a non-pushdown function runs, Flux pulls data into memory and runs all
+subsequent operations there.
+
+##### Pushdown functions in use
+```js
+from(bucket: "db/rp")
+  |> range(start: -1h)                       //
+  |> filter(fn: (r) => r.sensor == "abc123") //
+  |> group(columns: ["_field", "host"])      // Pushed to the data source
+  |> aggregateWindow(every: 5m, fn: max)     //
+  |> filter(fn: (r) => r._value >= 90.0)     //
+
+  |> top(n: 10)                              // Run in memory
+```
+
+### Avoid processing filters inline
+Avoid using mathematic operations or string manipulation inline to define data filters.
+Processing filter values inline prevents `filter()` from pushing its operation down
+to the underlying data source, so data returned by the
+previous function loads into memory.
+This often results in a significant performance hit.
+
+For example, the following query uses [dashboard variables](/influxdb/v2.0/visualize-data/variables/)
+and string concatenation to define a region to filter by.
+Because `filter()` uses string concatenation inline, it can't push its operation
+to the underlying data source and loads all data returned from `range()` into memory.
+
+```js
+from(bucket: "db/rp")
+  |> range(start: -1h)                      
+  |> filter(fn: (r) => r.region == v.provider + v.region)
+```
+
+To dynamically set filters and maintain the pushdown ability of the `filter()` function,
+use variables to define filter values outside of `filter()`:
+
+```js
+region = v.provider + v.region
+
+from(bucket: "db/rp")
+  |> range(start: -1h)                      
+  |> filter(fn: (r) => r.region == region)
+```
+
+## Avoid short window durations
+Windowing (grouping data based on time intervals) is commonly used to aggregate and downsample data.
+Increase performance by avoiding short window durations.
+More windows require more compute power to evaluate which window each row should be assigned to.
+Reasonable window durations depend on the total time range queried.
+
+## Use "heavy" functions sparingly
+The following functions use more memory or CPU than others.
+Consider their necessity in your data processing before using them:
+
+- [map()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/map/)
+- [reduce()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/aggregates/reduce/)
+- [join()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/join/)
+- [union()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/union/)
+- [pivot()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/pivot/)
+
+{{% note %}}
```

Comment:
should we remove this for Enterprise.... as we should advance this for each maintenance release and we know what the list is?